nlds/server_config.py

```diff
@@ -1,42 +1,58 @@
 import json
 import os.path
+
 from .nlds_setup import CONFIG_FILE_LOCATION
+from .utils.constants import CONFIG_SCHEMA
 
-def validate_config_file(json_config):
-    """Validate the JSON config file to match the schema in load_config_file."""
-    # Server section
-    try:
-        auth_section = json_config["authentication"]
-    except KeyError:
-        raise RuntimeError(
-            f"The config file at {CONFIG_FILE_LOCATION} does not contain an "
-            "['authentication'] section."
-        )
+def validate_config_file(json_config: dict) -> None:
+    """
+    Validate the JSON config file matches the schema defined in nlds_setup. 
+    Currently only checks that required headings and subheadings exist, i.e. only 
+    scans one layer deep and does no value checking. 
 
-    for key in ["authenticator_backend"]:
+    :param json_config:     Config file loaded using json.load()
+    
+    """
+    # Convert defined schema into a dictionary for ease of iteration
+    schema = dict(CONFIG_SCHEMA)
+
+    # Loop through and check that required headings and labels exist
+    for section_heading, section_labels in schema.items():
         try:
-            value = auth_section[key]
+            section = json_config[section_heading]
         except KeyError:
-            raise KeyError(
-                f"The config file at {CONFIG_FILE_LOCATION} does not "
-                f"contain {key} in the ['authentication'] section."
+            raise RuntimeError(
+                f"The config file at {CONFIG_FILE_LOCATION} does not contain a(n) "
+                f"['{section_heading}'] section."
             )
+        for sl in section_labels:
```

Comment:
I think this could be done more efficiently.
```if not sl in section_labels.keys():
    raise KeyError
```
Although the way you have done it might be more efficient!

Reply:
I think you're right but you still have to loop over the section_labels, unless you do a direct issubset comparison which might be slightly quicker, but at this size I'm not sure it matters a great deal?